packages/@n8n/db/src/migrations/common/1765886667897-AddAgentIdForeignKeys.ts

```diff
@@ -0,0 +1,50 @@
+import type { MigrationContext, ReversibleMigration } from '../migration-types';
+
+const table = {
+	agents: 'chat_hub_agents',
+	sessions: 'chat_hub_sessions',
+	messages: 'chat_hub_messages',
+} as const;
+
+export class AddAgentIdForeignKeys1765886667897 implements ReversibleMigration {
+	async up({ schemaBuilder: { addForeignKey }, runQuery, escape }: MigrationContext) {
+		// Clean up orphaned agentId references before adding foreign key constraint
+		await runQuery(
+			`UPDATE ${escape.tableName(table.sessions)} SET "agentId" = NULL WHERE "agentId" IS NOT NULL AND "agentId" NOT IN (SELECT id FROM ${escape.tableName(table.agents)})`,
+		);
+		await runQuery(
+			`UPDATE ${escape.tableName(table.messages)} SET "agentId" = NULL WHERE "agentId" IS NOT NULL AND "agentId" NOT IN (SELECT id FROM ${escape.tableName(table.agents)})`,
+		);
+
+		// Add foreign key constraint for agentId in sessions table
+		await addForeignKey(
+			table.sessions,
+			'agentId',
+			[table.agents, 'id'],
+			'FK_chat_hub_sessions_agentId',
+			'SET NULL',
```

Comment:
Why not cascade delete ? Is there a possibility that the agent is deleted and we should still keep the messages ?

Reply:
Yeah, seems better & more intuitive than deletion